scipy/stats/tests/test_morestats.py

```diff
@@ -2076,6 +2076,27 @@ def test_user_defined_optimizer_and_brack_raises_error(self):
             stats.boxcox_normmax(self.x, brack=(-2.0, 2.0),
                                  optimizer=optimizer)
 
+    def test_negative_ymax(self):
+        with pytest.raises(ValueError, match="`ymax` must be strictly positive"):
+            stats.boxcox_normmax(self.x, ymax=-1)
+
+    @pytest.mark.parametrize('x', [
+        # Attempt to trigger overflow in power expressions.
+        np.array([2003.0, 1950.0, 1997.0, 2000.0, 2009.0,
+                  2009.0, 1980.0, 1999.0, 2007.0, 1991.0]),
+        # Attempt to trigger overflow with a large optimal lambda.
+        np.array([2003.0, 1950.0, 1997.0, 2000.0, 2009.0]),
+        # Attempt to trigger overflow with large data.
+        np.array([2003.0e200, 1950.0e200, 1997.0e200, 2000.0e200, 2009.0e200])
+    ])
+    @pytest.mark.parametrize("ymax", [1e10, 1e30, 1e100])
+    # TODO: add method "pearsonr" after fix overflow issue
+    @pytest.mark.parametrize("method", ["mle"])
+    def test_user_defined_ymax(self, x, ymax, method):
```

Comment:
```suggestion
    def test_user_defined_ymax(self, x, ymax, method):
        # Use ymax=None and assert that there is indeed an overflow.
        # Then continue with the rest of this test.
```

Reply:
This was not meant to be merged. Just read the code suggestion.

Reply:
I would say, yes this also needs a test. But I'm curious myself what scipy maintainers say here.

Reply:
> Use ymax=None and assert that there is indeed an overflow.

I don't understand. If `ymax=None`, then it becomes a large positive number. There will not be an overflow to detect unless the user specifies a value greater than the maximum float of the given dtype.

> But just curious, what if there isn't overflow, but I want to specify `ymax` to ensure the transformed data small than `ymax`

Aren't you effectively checking that with
```python3
assert_allclose(ymax, ymax_res, rtol=1e-1)
```
?
I don't think we can guarantee that it will be *strictly* less due to numerical issues, so I think the `assert_allclose` as you have it is the best option.

Reply:
The logic here is to use data that indeed causes overflow (float64), then test `ymax` with several different values below the maximum float.

Actually, I tested `assert_allclose` with several `rtol`s, I choose 1e-1 is because other smaller values will raise an error.

Reply:
I understand that. The part that was was confusing was the comment "assert that there is indeed an overflow.". This meant "assert that a warning about overflow is emitted". There is no overflow that is directly detectable within the test; it is covered up within the function.

> Actually, I tested assert_allclose with several `rtol`s, I choose `1e-1` is because other smaller values will raise an error

Yes, I figured. Why do you mention it? I said it looks fine.

Re-reading your comment:

> what if there isn't overflow, but I want to specify `ymax` to ensure the transformed data small than `ymax`

It sounds like you want to add a test case with a different data set for which the optimal `lmbda` would not cause the transformed data to exceed `ymax`.  Sure, go ahead.